c10/cuda/CUDACachingAllocator.h

```diff
@@ -485,3 +485,52 @@ inline void enablePeerAccess(
 }
 
 } // namespace c10::cuda::CUDACachingAllocator
+
+namespace c10::cuda {
+
+// MemPool represents a pool of memory in a caching allocator. Currently,
+// it's just the ID of the pool object maintained in the CUDACachingAllocator.
```

Comment:
Can MemPools ever die or are they permanent and you never get to reclaim them?

Reply:
Since we're only ever passing raw pointers around I guess they never die

Reply:
I have not touched the actual PrivatePool objects that are owned by CUDACachingAllocator, so I believe their lifetime *will* be dictated by the allocator.